src/admin/wordlift_admin_save_post.php

```diff
@@ -101,3 +101,150 @@ function wl_update_post_status( $post_id, $status ) {
 	/** This action is documented in wp-includes/post.php */
 	do_action( 'wp_insert_post', $post->ID, $post, true );
 }
+
+/** This action to disable block editor*/
+add_filter( 'use_block_editor_for_post', '__return_false', 10 );
+
+/**
+ * Update the excerpt summary and create wl_summaries-excerpt meta key to hold API responce.
+ *
+ * @param string $location The location of post after save
+ * @param int $post_id post ID
+ */
+function update_post_excerpt_summary( $location, $post_id ) {
+    $rest = get_excerpt_summary($post_id);
+    $res = json_decode($rest, true);
+    if ($res) {
+        if (array_key_exists('summary', $res)) {
+            $summary = $res['summary'];
+            $postKey =  update_post_meta($post_id, "wl-summaries-excerpt", $summary);
+        }
+    } 
+    return $location;
+}
+add_filter( 'redirect_post_location', 'update_post_excerpt_summary', 10, 2 );
+
+/**
+ * Call summarize API.
+ *
+ * @param int $post_id The post ID
+ *
+ * @return JSON
+ */
+function get_excerpt_summary($post_id) {
+    $post = get_post($post_id);
+    $post_content = $post->post_content;
+    $curl = curl_init();
+    curl_setopt_array($curl, array(
```

Comment:
@ravisoni6262 we shouldn't use CURL directly here, we should rely on WP's functions.